packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js

```diff
@@ -65,19 +71,47 @@ describe('Marketplace page', () => {
   afterAll(() => server.close());
 
   it('renders and matches the plugin tab snapshot', async () => {
+    // Check snapshot
     const { container, getByTestId, getByRole } = render(App);
     await waitForElementToBeRemoved(() => getByTestId('loader'));
     await waitFor(() => expect(getByRole('heading', { name: /marketplace/i })).toBeInTheDocument());
 
     expect(container.firstChild).toMatchSnapshot();
+
+    // Make sure it defaults to the plugins tab
+    const button = screen.getByRole('tab', { selected: true });
+    const pluginsTabActive = getByText(button, /plugins/i);
+
+    const tabPanel = screen.getByRole('tabpanel');
+    const pluginCardText = getByText(tabPanel, 'Comments');
+    const providerCardText = queryByText(tabPanel, 'Cloudinary');
+    const submitPluginText = queryByText(container, 'Submit plugin');
+
+    expect(pluginsTabActive).not.toBe(null);
+    expect(pluginCardText).toBeVisible();
+    expect(submitPluginText).toBeVisible();
+    expect(providerCardText).toEqual(null);
   });
 
-  it('renders and matches the provider tab snapshot', async () => {
+  it('renders and matches the provider tab snapshot', () => {
+    // Make sure it switches to the providers tab
     const { container, getByRole } = render(App);
-    await waitFor(() => expect(getByRole('heading', { name: /marketplace/i })).toBeInTheDocument());
-    const providersTab = screen.getByRole('tab', { selected: false });
+    const providersTab = getByRole('tab', { name: /providers/i });
     fireEvent.click(providersTab);
+    const button = getByRole('tab', { selected: true });
+    const providersTabActive = getByText(button, /Providers/i);
 
+    const tabPanel = getByRole('tabpanel');
+    const providerCardText = getByText(tabPanel, 'Cloudinary');
+    const pluginCardText = queryByText(tabPanel, 'Comments');
+    const submitProviderText = queryByText(container, 'Submit provider');
+
+    expect(providersTabActive).not.toBe(null);
+    expect(providerCardText).toBeVisible();
+    expect(submitProviderText).toBeVisible();
+    expect(pluginCardText).toEqual(null);
```

Comment:
I don't really understand this change for providers and plugins, what's the added benefit? 

I think it's normal to have similar state setups in tests and it is better to test specific things in isolation. Here we test the snapshot, and fireEvent to test the providers tab is set to active, and test the correct content is displayed after that event.

Reply:
The problem I see is that it's not just similar test setups, our tests literally share the same state. Updating it in one test affects the other. So previously testing the "default" tab only worked because the previous test did a click on the plugins tab.

But it's true the change a made is more of a workaround than anything. I can look into making sure the state is reset between each test

Reply:
oh ok gotcha, maybe we could organize them into describe suites for plugins and providers and do some beforeEach magic to make sure each test has a clean state (another PR). But works for me thanks for explaining.